### ISSUE TITLE
invalid SSL cert for `*.packages.shiftkey.dev`

### ISSUE THREAD

**Issue by nuernbergerA** [shiftkey/desktop#1062]
Hey @shiftkey,

it seems that your ssl cert for https://apt.packages.shiftkey.dev/ is the default azure one and causes an error for `apt`

> Error: https://apt.packages.shiftkey.dev/ubuntu any InRelease
>   Certificate verification failed: The certificate is NOT trusted. The name in the certificate does not match the expected.  Could not handshake: Error in the certificate verification. [IP: 13.107.246.44 443]

**Comment by krushnaj** [comment]
Facing a similar issue when installing GitHub Desktop using @shiftkey package feed

**Comment by GlacierFox** [comment]
Similar issue on RHEL too.

```
Errors during downloading metadata for repository 'shiftkey-packages':
  - Curl error (60): SSL peer certificate or SSH remote key was not OK for https://rpm.packages.shiftkey.dev/rpm/repodata/repomd.xml [SSL: no alternative certificate subject name matches target host name 'rpm.packages.shiftkey.dev']
Error: Failed to download metadata for repo 'shiftkey-packages': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried
```

**Comment by KBeDevel** [comment]
Same issue in RPM repo

**Comment by RogueScholar** [comment]
Just thought I would add that "invalid" in this case does not mean "expired," as tends to be the case when TLS goes awry and is what I expected to find when I myself encountered this issue just now. Rather, the Subject (Common Name) of the certificate no longer matches the domain name, instead being issued for `azureedge.net` rather than `*.packages.shiftkey.dev`. As the domain is also enrolled in HTST (good choice, BTW), the subject mismatch has completely disabled access to the domain. Hopefully this helps you to dig right into the meat of the problem when you have the time to troubleshoot it.

Thanks for maintaining such a useful resource for us Linux diehards out there. 👍🏻

**Comment by Tiagoquix** [comment]
Similar issue on Fedora Linux 40:

```bash
GitHub Desktop                                                                0.0  B/s |   0  B     00:00    
Errors during downloading metadata for repository 'shiftkey-packages':
  - Curl error (60): SSL peer certificate or SSH remote key was not OK for https://rpm.packages.shiftkey.dev/rpm/repodata/repomd.xml [SSL: no alternative certificate subject name matches target host name 'rpm.packages.shiftkey.dev']
Error: Failed to download metadata for repo 'shiftkey-packages': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried
Ignoring repositories: shiftkey-packages
```

**Comment by BrycensRanch** [comment]
its joever 😭

**Comment by nikosandreou** [comment]
Same here, Ubuntu 22.04.4 LTS. As @RogueScholar said, if you try to open the link in a browser (Firefox in my case), you see the following:

> apt.packages.shiftkey.dev has a security policy called HTTP Strict Transport Security (HSTS), which means that Firefox can only connect to it securely. You can’t add an exception to visit this site.
> 
> The issue is most likely with the website, and there is nothing you can do to resolve it. You can notify the website’s administrator about the problem.
> 
> Websites prove their identity via certificates. Firefox does not trust this site because it uses a certificate that is not valid for apt.packages.shiftkey.dev. The certificate is only valid for the following names: *.azureedge.net, *.media.microsoftstream.com, *.origin.mediaservices.windows.net, *.streaming.mediaservices.windows.net
>  
> Error code: [SSL_ERROR_BAD_CERT_DOMAIN]

**Comment by mei10101** [comment]
I have same issue on Fedora 40 
`Errors during downloading metadata for repository 'shiftkey-packages':
  - Curl error (60): SSL peer certificate or SSH remote key was not OK for https://rpm.packages.shiftkey.dev/rpm/repodata/repomd.xml [SSL: no alternative certificate subject name matches target host name 'rpm.packages.shiftkey.dev']
Error: Failed to download metadata for repo 'shiftkey-packages': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried`

**Comment by Gaudenz77** [comment]
Hi there, i tried to do a simple sudo apt update && upgrade today and apt.package.shiftkey.dev denied being on a trusted certificate. here is a clip from my terminal. 
do i need to fix/do/un-reinstall anything? Im not that deep into it to know exactly how to challenge such a problem, but it disturbs my otherwise fine working update and upgrade view and process. Tell me if you need more info.
The github-dektop is installed within 
- Virtualbox@latest
- Guest Ubuntu 22.04@lateset
- Windows 11 hoste @latest
- Asus PC
as mentioned, if more info needed, feel free to ask. looking out for a fix. Workaround in the certification repo failed category didnt help much...
thx in advance, and i like github dektop on ubunt much, thx 4 that app port...:
```
Hit:9 https://ppa.launchpadcontent.net/flatpak/stable/ubuntu jammy InRelease                            
Hit:10 https://ppa.launchpadcontent.net/ondrej/apache2/ubuntu jammy InRelease                           
Hit:11 https://ppa.launchpadcontent.net/ondrej/php/ubuntu jammy InRelease                               
Get:1 https://packages.microsoft.com/repos/code stable InRelease [3’590 B]                              
Ign:7 https://apt.packages.shiftkey.dev/ubuntu any InRelease                             
Ign:7 https://apt.packages.shiftkey.dev/ubuntu any InRelease
Err:7 https://apt.packages.shiftkey.dev/ubuntu any InRelease                                            
  Certificate verification failed: The certificate is NOT trusted. The name in the certificate does not match the expected.  Could not handshake: Error in the certificate verification. [IP: 13.107.213.60 443]
Fetched 114 kB in 8s (13.5 kB/s)                                                                        
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
All packages are up to date.
W: Failed to fetch https://apt.packages.shiftkey.dev/ubuntu/dists/any/InRelease  Certificate verification failed: The certificate is NOT trusted. The name in the certificate does not match the expected.  Could not handshake: Error in the certificate verification. [IP: 13.107.213.60 443]
W: Some index files failed to download. They have been ignored, or old ones used instead.
krakatoom@krakatoom-Desktop:~/Downloads$ ```

**Comment by wolfy1339** [comment]
Please don't post me too comments...

There is nothing to be done on the client side. We have to wait for @shiftkey to fix thie

**Comment by Gaudenz77** [comment]
> Please don't post me too comments...
> 
> There is nothing to be done on the client side. We have to wait for @shiftkey to fix thie

thanks for info. I am  rather new to this, the main info was more my terminal sequence, i thought it could help. But I am glad to get more infos on what or how I can better interact in places like this. so if you can give me some advice, i gladly take it.

**Comment by shiftkey** [comment]
:wave: Apologies for the delay - I've been on holidays the past couple of weeks so this had to sit broken.

This has now been restored and I'll do a bit of a write-up on the next steps when I have bandwidth to document this further.

**Comment by KBeDevel** [comment]
Thank you @shiftkey! RPM repo is working again (at least for Fedora 40)

**Comment by tstraus13** [comment]
Unsure if related but I am getting timeouts with DNF on Fedora on the rpm.packages.shiftkey.dev. I have been getting them the past few days.

**Comment by KBeDevel** [comment]
> Unsure if related but I am getting timeouts with DNF on Fedora on the rpm.packages.shiftkey.dev. I have been getting them the past few days.

Which DNS are you using? I cannot reproduce the issue (using Fedora 40 too).

**Comment by tstraus13** [comment]
> > Unsure if related but I am getting timeouts with DNF on Fedora on the rpm.packages.shiftkey.dev. I have been getting them the past few days.
> 
> Which DNS are you using? I cannot reproduce the issue (using Fedora 40 too).

Thanks. Looks like it was DNS. I use PiHole but it was not blocking the request. Any outside requests are forwarded to Cloudflare DNS and for some reason I was not getting through. If I switched to another public DNS it seems to work. Thanks!

**Comment by KBeDevel** [comment]
Certificate is failing again in the RPM repo

**Comment by Tiagoquix** [comment]
Yep, the same issue has been revived.

**Comment by user97116** [comment]
Facing same issue on Ubuntu 24.04 
```
: Failed to fetch https://apt.packages.shiftkey.dev/ubuntu/dists/any/InRelease  Certificate verification failed: The certificate is NOT trusted. The name in the certificate does not match the expected.  Could not handshake: Error in the certificate verification. 
```

**Comment by shiftkey** [comment]
I haven't done this write-up yet so I'm gonna reopen this for now to remind myself to not forget this issue...

**Comment by tim-littlefair** [comment]
Hi @shiftkey, there was a post on reddit which wrongly claimed you were difficult to contact so I raised this concern on the upstream repo instead, that is the ticket at https://github.com/desktop/desktop/issues/18963 which @sergiou87 closed about a day ago.

I won't repeat the whole ticket here, but this is the key to the problem:

![image](https://github.com/user-attachments/assets/f723877d-3e70-4f69-911a-6edcc73429f1)

Please let me know if there is anything I can do at my end to help with the diagnosis (e.g. if you suspect that my DNS is not returning the same IP as other DNS at other locations where the problem does not arise).

**Comment by tim-littlefair** [comment]
Just did a fresh 'apt update', got this response:
W: Failed to fetch https://apt.packages.shiftkey.dev/ubuntu/dists/any/InRelease  
Cannot initiate the connection to apt.packages.shiftkey.dev:443 (2620:1ec:bdf::31). - connect (101: Network is unreachable) 
Could not connect to apt.packages.shiftkey.dev:443 (13.107.246.31), connection timed out

So resolving to different IPv6 and IPv4 addresses (unsurprising if this is hosted on Azure), now timing out rather than returning a bad certificate.  

BTW apologies for raising the completely unfounded supply chain suspicion - I was panicking because azureedge.net and microsoftstream.com looked to me like domains which I might see in a phishing scam, but I should have realized that windows.net was unlikely to be under hostile control.

**Comment by zaid1442011** [comment]
Have the same issue in the Debian repository.

**Comment by IainMcCracken** [comment]
Aug 11, 2024 here. CN on the cert when I go to apt.packages.shiftkey.dev is "*.azureedge.net", so apt rejects it, and it is not possible to update/upgrade/anything.

**Comment by IntrovertedFL** [comment]
Same issues Pop OS 22.04

**Comment by gogoSpace** [comment]
Using gdebi for now to install latest update. Thanks for your work. Anyway, do you have any plans to fix this in the near future?

**Comment by mwt** [comment]
Feel free to steer more traffic to my mirror to reduce bandwidth usage.

The most extreme would be to remove the source repo from the readme and only list it as the source for gpg keys.

(For security context, I don't have shiftkey's private key. I just download and serve up the files that he has signed.)

**Comment by samuel-andres** [comment]
```sh
$ sudo dnf upgrade
GitHub Desktop                                                                                                                                                                                         0.0  B/s |   0  B     00:00
Errors during downloading metadata for repository 'shiftkey-packages':
  - Curl error (60): SSL peer certificate or SSH remote key was not OK for https://rpm.packages.shiftkey.dev/rpm/repodata/repomd.xml [SSL: no alternative certificate subject name matches target host name 'rpm.packages.shiftkey.dev']
Error: Failed to download metadata for repo 'shiftkey-packages': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried
Ignoring repositories: shiftkey-packages
Last metadata expiration check: 2:45:14 ago on Mon 11 Nov 2024 08:18:29 PM -03.
Dependencies resolved.
Nothing to do.
Complete!

```

**Comment by benlambert43** [comment]
December 13th here. Looks like an another certificate problem:

`W: Failed to fetch https://apt.packages.shiftkey.dev/ubuntu/dists/any/InRelease  Certificate verification failed: The certificate is NOT trusted. The name in the certificate does not match the expected.  Could not handshake: Error in the certificate verification. [IP: 13.107.246.71 443]
`

**Comment by wolfy1339** [comment]
February 11
```
>>> Curl error (60): SSL peer certificate or SSH remote key was not OK for https://rpm.packages.shiftkey.dev/rpm/repodata/repomd.xml [SSL: no alternative certificate subject name matches target hostname 'rpm.packages.shiftkey.dev']```

**Comment by Margaret6556** [comment]
SSL issue certifcate NOT matching from Fedora41 

![Image](https://github.com/user-attachments/assets/c874b05e-f1e8-4255-a146-dc2b2587487e)

Server certificate
subject=C=US, ST=WA, L=Redmond, O=Microsoft Corporation, CN=*.azureedge.net
issuer=C=US, O=Microsoft Corporation, CN=Microsoft Azure RSA TLS Issuing CA 07


I tried to debug 2x, no success.  Pls advise and fix. Thanks

**Comment by tylerebowers** [comment]
Getting SSL errors here too, the common name is incorrect?
Mint 22.1
![Image](https://github.com/user-attachments/assets/37fbdd0a-c4b4-4810-9919-7e7711877d3b)
I have disabled the repo for now.

**Comment by Margaret6556** [comment]
> Getting SSL errors here too, the common name is incorrect? Mint 22.1 ![Image](https://github.com/user-attachments/assets/37fbdd0a-c4b4-4810-9919-7e7711877d3b) I have disabled the repo for now.

@tylerebowers Hi, Why did you disabled the repo? What are the pros and cons?

**Comment by tylerebowers** [comment]
> > Getting SSL errors here too, the common name is incorrect? Mint 22.1 ![Image](https://github.com/user-attachments/assets/37fbdd0a-c4b4-4810-9919-7e7711877d3b) I have disabled the repo for now.
> 
> [@tylerebowers](https://github.com/tylerebowers) Hi, Why did you disabled the repo? What are the pros and cons?

I only use shiftkey for github-desktop. I won't get updates while it is disabled. The pros are I can update my system now.

**Comment by Margaret6556** [comment]
I see. Then, you can enable it once fixed. Ok, I'l probably do the same. Thanks! Cool, portfolio, btw.

**Comment by mwt** [comment]
@tylerebowers @Margaret6556 It ran out of bandwidth. You can use the mwt repo in the readme.

**Comment by wolfy1339** [comment]
Hi @mwt, it seems that your mirror is missing the latest release

https://github.com/shiftkey/desktop/releases/tag/release-3.4.13-linux1

While your mirror stops at 3.4.12

**Comment by mwt** [comment]
@wolfy1339 said:
> Hi [@mwt](https://github.com/mwt), it seems that your mirror is missing the latest release

It'll update when I can pull the update from shiftkey's repo.

I cannot manually add updates to my mirror because I do not have shiftkey's private keys. He signs the repos and I redistribute his signature as-is.

**Comment by MaxEtMoritz** [comment]
upon (temporarily) disabling HSTS in my browser i got this:

![Image](https://github.com/user-attachments/assets/0a3d135f-ecb9-40d3-858f-0401e651c54e)

**Comment by shitwolfymakes** [comment]
Having this issue on 24.04 as well. Running the [shiftkey commands](https://github.com/shiftkey/desktop?tab=readme-ov-file#shiftkey-package-feed) results in:
```
gpg: no valid OpenPGP data found.
```

**Comment by wolfy1339** [comment]
Can we please lock this issue to reduce the spam?

To people reading this comment, the reason the repo isn't working is because the Azure instance runs out of bandwidth, and because of that the HTTPs certificate is changed to the default Azure one and not the one for shiftkey.

Use the MWT mirror as described in the README.

TL;DR this is a known issue and posting here won't help.

**Comment by Pantoush** [comment]
Confirming this is happening again as of July 27, 2026 on Ubuntu 24.04 (noble). Same pattern as before:

subject=C=US, ST=WA, L=Redmond, O=Microsoft Corporation, CN=*.azureedge.net
issuer=C=US, O=Microsoft Corporation, CN=Microsoft Azure TLS Issuing CA

apt update fails with the expected cert-mismatch error. Switched to the MWT mirror in the meantime — thanks for maintaining this.

**Comment by QuentiumYT** [comment]
This fork of GitHub Desktop is no longer active, I'd recommend you to use https://desktop-plus.org/ :)
